src/coreComponents/constitutive/capillaryPressure/JFunctionCapillaryPressure.cpp

```diff
@@ -250,6 +250,7 @@ void JFunctionCapillaryPressure::saveConvergedRockState( arrayView2d< real64 con
     {
       permeability = convergedPermeability[ei][0][2];
     }
+    GEOS_ERROR_IF( permeability < LvArray::NumericLimits< real64 >::epsilon, GEOS_FMT( "{}: zero permeability in J-function capillary pressure", getFullName() ));
```

Comment:
@paveltomin We generally cannot have `GEOS_*` macros inside a device lambda.

Reply:
this is on me. Isn't this calling `LVARRAY_ERROR` behind the scenes ? That one should work fine on device.  not sure about the FMT part though.

Reply:
Here: https://github.com/GEOS-DEV/LvArray/blob/0a5f594934e428548917376aeb6d3691c3b9206b/src/Macros.hpp#L123

Reply:
> Here: https://github.com/GEOS-DEV/LvArray/blob/0a5f594934e428548917376aeb6d3691c3b9206b/src/Macros.hpp#L123

so it will use print and it will work but it will have an impact on performance. It's probably okay for this specific kernel.

Reply:
Well then...`GEOS_FMT` cannot be called aside a device lambda.

Reply:
I will pull it out, no problem
Let me move that PR aside for now

Reply:
I think you can just use a string literal and it will work.

Reply:
ok, sounds good, done